advanced-achievements-plugin/src/main/java/com/hm/achievement/utils/CommandUtils.java

```diff
@@ -0,0 +1,816 @@
+package com.hm.achievement.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.bukkit.Bukkit;
+import org.bukkit.GameMode;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Damageable;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.entity.minecart.CommandMinecart;
+import org.bukkit.scoreboard.Objective;
+import org.bukkit.scoreboard.Team;
+
+import org.bukkit.Bukkit;
+import org.bukkit.GameMode;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Damageable;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.entity.minecart.CommandMinecart;
+import org.bukkit.scoreboard.Objective;
+import org.bukkit.scoreboard.Team;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class CommandUtils {
+
+	/**
+	 * The CommandUtils class is used for selector support in commands (@p, @a etc.) Originally made by ZombieStriker
```

Comment:
Good practice to have left this notice here! Though I _think_ we need to go a little further, given that these files are licensed under GPL. I suggest the following:
* only link only to one of them as they use different versions of the GPL license.
* move the class to it own package (maybe `com.hm.achievement.command.external`?).
* version control the license file (either https://github.com/ZombieStriker/PsudoCommands/blob/master/LICENSE or https://github.com/ZombieStriker/CommandUtils/blob/master/LICENSE) alongside the class.

This will make it clearer that it's a third-party file and will cover the "include original license" requirement of GPL.